src/cpp/server_core/ServerDatabase.cpp

```diff
@@ -151,7 +151,23 @@ Error readOptions(const std::string& databaseConfigFile,
          // always ensure the database file user is correct, if specified
          error = databaseFile.changeOwnership(databaseFileUser.get());
          if (error)
+         {
+            error.addProperty("description", "Unable to change ownership of database file");
             return error;
+         }
+
+#ifndef _WIN32
+         error = databaseDirectory.changeOwnership(databaseFileUser.get());
+         if (error)
+         {
+            bool writable = false;
+            Error writableError = databaseDirectory.isWriteable(writable);
+            if (writableError || !writable)
+            {
+                LOG_ERROR_MESSAGE("Sqllite database directory: " + databaseDirectory.getAbsolutePath() + " must be writable by: " + databaseFileUser.get().getUsername());
```

Comment:
Shouldn't we exit with an error after logging this message?

Reply:
For OS, it seems like things work pretty well even with a read-only database. I was concerned about adding a new "exit error" that might trip up some users that don't care about the DB.

Reply:
I see, that seems reasonable.